### ISSUE TITLE
pip install -r framework/requirements.txt fails on "clean" installations

### ISSUE THREAD

**Issue by mhamra** [wazuh/wazuh#16373]
|wazuh branch | component|
|---|---|
| 4.4 and 4.5 | framework/requirements.txt|

## Description
Starting from a docker container with:
- "clean" ubuntu 20.04 docker container 
- Python 3.9 on the container
- a copy of the wazuh repository on branch 4.4 or 4.5
- a virtual environment installed and activated on the wazuh folder 

the following command fails:
```
(wazuh-4.4) root@bc4bbfd2e45f:~/wazuh-4.4#pip install -r framework/requirements.txt 
```
pip command output:

```(wazuh-4.4) root@bc4bbfd2e45f:~/wazuh-4.4# pip install -r framework/requirements.txt
Collecting aiohttp==3.8.1
  Using cached aiohttp-3.8.1-cp39-cp39-manylinux_2_5_x86_64.manylinux1_x86_64.manylinux_2_12_x86_64.manylinux2010_x86_64.whl (1.2 MB)
Collecting aiohttp-cache==2.2.0
  Using cached aiohttp_cache-2.2.0-py3-none-any.whl (7.8 kB)
Collecting aiohttp-cors==0.7.0
  Using cached aiohttp_cors-0.7.0-py3-none-any.whl (27 kB)
Collecting aiohttp-jinja2==1.4.2
  Using cached aiohttp_jinja2-1.4.2-py3-none-any.whl (11 kB)
Collecting aioredis==1.3.1
  Using cached aioredis-1.3.1-py3-none-any.whl (65 kB)
Collecting aiosignal==1.2.0
  Using cached aiosignal-1.2.0-py3-none-any.whl (8.2 kB)
Requirement already satisfied: asn1crypto==1.3.0 in ./lib/python3.9/site-packages (from -r framework/requirements.txt (line 7)) (1.3.0)
Requirement already satisfied: async-timeout==4.0.2 in ./lib/python3.9/site-packages (from -r framework/requirements.txt (line 8)) (4.0.2)
Collecting attrs==20.3.0
  Using cached attrs-20.3.0-py2.py3-none-any.whl (49 kB)
Requirement already satisfied: azure-common==1.1.25 in ./lib/python3.9/site-packages (from -r framework/requirements.txt (line 10)) (1.1.25)
Collecting azure-storage-blob==2.1.0
  Using cached azure_storage_blob-2.1.0-py2.py3-none-any.whl (88 kB)
Collecting azure-storage-common==2.1.0
  Using cached azure_storage_common-2.1.0-py2.py3-none-any.whl (47 kB)
Collecting boto3==1.17.85
  Using cached boto3-1.17.85-py2.py3-none-any.whl (131 kB)
Collecting botocore==1.20.85
  Using cached botocore-1.20.85-py2.py3-none-any.whl (7.6 MB)
Collecting cachetools==4.1.0
  Using cached cachetools-4.1.0-py3-none-any.whl (10 kB)
Collecting certifi==2022.12.7
  Using cached certifi-2022.12.7-py3-none-any.whl (155 kB)
Collecting cffi==1.14.4
  Using cached cffi-1.14.4-cp39-cp39-manylinux1_x86_64.whl (405 kB)
Requirement already satisfied: chardet==3.0.4 in ./lib/python3.9/site-packages (from -r framework/requirements.txt (line 18)) (3.0.4)
Collecting charset-normalizer==2.0.4
  Using cached charset_normalizer-2.0.4-py3-none-any.whl (36 kB)
Collecting click==8.1.3
  Using cached click-8.1.3-py3-none-any.whl (96 kB)
Collecting clickclick==20.10.2
  Using cached clickclick-20.10.2-py2.py3-none-any.whl (7.4 kB)
Collecting connexion==2.6.0
  Using cached connexion-2.6.0-py2.py3-none-any.whl (76 kB)
Collecting cryptography==3.3.2
  Using cached cryptography-3.3.2-cp36-abi3-manylinux2010_x86_64.whl (2.6 MB)
Collecting Cython==0.29.21
  Using cached Cython-0.29.21-cp39-cp39-manylinux1_x86_64.whl (1.9 MB)
Collecting defusedxml==0.6.0
  Using cached defusedxml-0.6.0-py2.py3-none-any.whl (23 kB)
Collecting docker==4.2.0
  Using cached docker-4.2.0-py2.py3-none-any.whl (143 kB)
Collecting docker-pycreds==0.4.0
  Using cached docker_pycreds-0.4.0-py2.py3-none-any.whl (9.0 kB)
Collecting docutils==0.15.2
  Using cached docutils-0.15.2-py3-none-any.whl (547 kB)
Collecting ecdsa==0.16.1
  Using cached ecdsa-0.16.1-py2.py3-none-any.whl (104 kB)
Requirement already satisfied: envparse==0.2.0 in ./lib/python3.9/site-packages (from -r framework/requirements.txt (line 30)) (0.2.0)
Collecting Flask==2.0.0
  Using cached Flask-2.0.0-py3-none-any.whl (93 kB)
Collecting frozenlist==1.2.0
  Using cached frozenlist-1.2.0-cp39-cp39-manylinux_2_5_x86_64.manylinux1_x86_64.manylinux_2_12_x86_64.manylinux2010_x86_64.whl (203 kB)
Collecting future==0.18.3
  Using cached future-0.18.3-py3-none-any.whl
Collecting google-api-core==1.30.0
  Using cached google_api_core-1.30.0-py2.py3-none-any.whl (93 kB)
Collecting googleapis-common-protos==1.51.0
  Using cached googleapis_common_protos-1.51.0-py3-none-any.whl
Collecting google-auth==1.28.0
  Using cached google_auth-1.28.0-py2.py3-none-any.whl (136 kB)
Collecting google-cloud-core==1.7.1
  Using cached google_cloud_core-1.7.1-py2.py3-none-any.whl (28 kB)
Collecting google-cloud-pubsub==2.7.1
  Using cached google_cloud_pubsub-2.7.1-py2.py3-none-any.whl (217 kB)
Collecting google-cloud-storage==1.39.0
  Using cached google_cloud_storage-1.39.0-py2.py3-none-any.whl (103 kB)
Collecting google-crc32c==1.1.2
  Using cached google_crc32c-1.1.2-cp39-cp39-manylinux2014_x86_64.whl (38 kB)
Collecting google-resumable-media==1.3.1
  Using cached google_resumable_media-1.3.1-py2.py3-none-any.whl (75 kB)
Collecting grpc-google-iam-v1==0.12.3
  Using cached grpc_google_iam_v1-0.12.3-py3-none-any.whl
Collecting grpcio==1.38.1
  Using cached grpcio-1.38.1-cp39-cp39-manylinux2014_x86_64.whl (4.3 MB)
Collecting hiredis==1.1.0
  Using cached hiredis-1.1.0.tar.gz (54 kB)
  Preparing metadata (setup.py) ... done
Requirement already satisfied: idna==2.9 in ./lib/python3.9/site-packages (from -r framework/requirements.txt (line 45)) (2.9)
Requirement already satisfied: inflection==0.3.1 in ./lib/python3.9/site-packages (from -r framework/requirements.txt (line 46)) (0.3.1)
Requirement already satisfied: itsdangerous==2.0.0 in ./lib/python3.9/site-packages (from -r framework/requirements.txt (line 47)) (2.0.0)
Collecting Jinja2==3.0.0
  Using cached Jinja2-3.0.0-py3-none-any.whl (133 kB)
Requirement already satisfied: jmespath==0.9.5 in ./lib/python3.9/site-packages (from -r framework/requirements.txt (line 49)) (0.9.5)
Requirement already satisfied: jsonschema==2.6.0 in ./lib/python3.9/site-packages (from -r framework/requirements.txt (line 50)) (2.6.0)
Collecting libcst==0.3.20
  Using cached libcst-0.3.20-py3-none-any.whl (514 kB)
Requirement already satisfied: MarkupSafe==2.1.2 in ./lib/python3.9/site-packages (from -r framework/requirements.txt (line 52)) (2.1.2)
Requirement already satisfied: more-itertools==8.2.0 in ./lib/python3.9/site-packages (from -r framework/requirements.txt (line 53)) (8.2.0)
Requirement already satisfied: multidict==5.1.0 in ./lib/python3.9/site-packages (from -r framework/requirements.txt (line 54)) (5.1.0)
Requirement already satisfied: mypy-extensions==0.4.3 in ./lib/python3.9/site-packages (from -r framework/requirements.txt (line 55)) (0.4.3)
Collecting openapi-spec-validator==0.2.6
  Using cached openapi_spec_validator-0.2.6-py3-none-any.whl (23 kB)
Collecting packaging==20.9
  Using cached packaging-20.9-py2.py3-none-any.whl (40 kB)
Requirement already satisfied: pathlib==1.0.1 in ./lib/python3.9/site-packages (from -r framework/requirements.txt (line 58)) (1.0.1)
Requirement already satisfied: protobuf==3.19.6 in ./lib/python3.9/site-packages (from -r framework/requirements.txt (line 59)) (3.19.6)
Collecting proto-plus==1.19.0
  Using cached proto_plus-1.19.0-py3-none-any.whl (42 kB)
Requirement already satisfied: psutil==5.8.0 in ./lib/python3.9/site-packages (from -r framework/requirements.txt (line 61)) (5.8.0)
Requirement already satisfied: pyasn1==0.4.8 in ./lib/python3.9/site-packages (from -r framework/requirements.txt (line 62)) (0.4.8)
Requirement already satisfied: pyasn1-modules==0.2.8 in ./lib/python3.9/site-packages (from -r framework/requirements.txt (line 63)) (0.2.8)
Requirement already satisfied: pycparser==2.20 in ./lib/python3.9/site-packages (from -r framework/requirements.txt (line 64)) (2.20)
Requirement already satisfied: pyparsing==2.4.7 in ./lib/python3.9/site-packages (from -r framework/requirements.txt (line 65)) (2.4.7)
Collecting python-dateutil==2.8.1
  Using cached python_dateutil-2.8.1-py2.py3-none-any.whl (227 kB)
Collecting python-jose==3.1.0
  Using cached python_jose-3.1.0-py2.py3-none-any.whl (26 kB)
Requirement already satisfied: python-json-logger==2.0.2 in ./lib/python3.9/site-packages (from -r framework/requirements.txt (line 68)) (2.0.2)
Requirement already satisfied: pytz==2020.1 in ./lib/python3.9/site-packages (from -r framework/requirements.txt (line 69)) (2020.1)
Requirement already satisfied: PyYAML==5.4.1 in ./lib/python3.9/site-packages (from -r framework/requirements.txt (line 70)) (5.4.1)
Collecting requests==2.25.1
  Using cached requests-2.25.1-py2.py3-none-any.whl (61 kB)
Requirement already satisfied: rsa==4.7.2 in ./lib/python3.9/site-packages (from -r framework/requirements.txt (line 72)) (4.7.2)
Collecting s3transfer==0.4.2
  Using cached s3transfer-0.4.2-py2.py3-none-any.whl (79 kB)
Requirement already satisfied: secure==0.2.1 in ./lib/python3.9/site-packages (from -r framework/requirements.txt (line 74)) (0.2.1)
Requirement already satisfied: six==1.14.0 in ./lib/python3.9/site-packages (from -r framework/requirements.txt (line 75)) (1.14.0)
Requirement already satisfied: SQLAlchemy==1.3.11 in ./lib/python3.9/site-packages (from -r framework/requirements.txt (line 76)) (1.3.11)
Requirement already satisfied: tabulate==0.8.9 in ./lib/python3.9/site-packages (from -r framework/requirements.txt (line 77)) (0.8.9)
Requirement already satisfied: typing-extensions==3.10.0.2 in ./lib/python3.9/site-packages (from -r framework/requirements.txt (line 78)) (3.10.0.2)
Requirement already satisfied: typing-inspect==0.7.1 in ./lib/python3.9/site-packages (from -r framework/requirements.txt (line 79)) (0.7.1)
Requirement already satisfied: urllib3==1.26.5 in ./lib/python3.9/site-packages (from -r framework/requirements.txt (line 80)) (1.26.5)
Requirement already satisfied: uvloop==0.15.2 in ./lib/python3.9/site-packages (from -r framework/requirements.txt (line 81)) (0.15.2)
Collecting websocket-client==0.57.0
  Using cached websocket_client-0.57.0-py2.py3-none-any.whl (200 kB)
Collecting Werkzeug==2.2.3
  Using cached Werkzeug-2.2.3-py3-none-any.whl (233 kB)
Requirement already satisfied: xmltodict==0.12.0 in ./lib/python3.9/site-packages (from -r framework/requirements.txt (line 84)) (0.12.0)
Collecting yarl==1.6.3
  Using cached yarl-1.6.3-cp39-cp39-manylinux2014_x86_64.whl (315 kB)
Requirement already satisfied: setuptools>=40.3.0 in ./lib/python3.9/site-packages (from google-api-core==1.30.0->-r framework/requirements.txt (line 34)) (58.1.0)
Collecting google-api-core[grpc]<3.0.0dev,>=1.26.0
  Using cached google_api_core-2.11.0-py3-none-any.whl (120 kB)
Collecting googleapis-common-protos[grpc]<2.0.0dev,>=1.5.2
  Using cached googleapis_common_protos-1.58.0-py2.py3-none-any.whl (223 kB)
Collecting google-api-core[grpc]<3.0.0dev,>=1.26.0
  Using cached google_api_core-2.10.2-py3-none-any.whl (115 kB)
  Using cached google_api_core-2.10.1-py3-none-any.whl (115 kB)
  Using cached google_api_core-2.10.0-py3-none-any.whl (115 kB)
  Using cached google_api_core-2.9.0-py3-none-any.whl (115 kB)
  Using cached google_api_core-2.8.2-py3-none-any.whl (114 kB)
  Using cached google_api_core-2.8.1-py3-none-any.whl (114 kB)
  Using cached google_api_core-2.8.0-py3-none-any.whl (114 kB)
  Using cached google_api_core-2.7.3-py3-none-any.whl (114 kB)
  Using cached google_api_core-2.7.2-py3-none-any.whl (114 kB)
  Using cached google_api_core-2.7.1-py3-none-any.whl (114 kB)
  Using cached google_api_core-2.7.0-py3-none-any.whl (114 kB)
  Using cached google_api_core-2.6.1-py3-none-any.whl (114 kB)
  Using cached google_api_core-2.6.0-py2.py3-none-any.whl (114 kB)
  Using cached google_api_core-2.5.0-py2.py3-none-any.whl (111 kB)
  Using cached google_api_core-2.4.0-py2.py3-none-any.whl (111 kB)
  Using cached google_api_core-2.3.2-py2.py3-none-any.whl (109 kB)
  Using cached google_api_core-2.3.0-py2.py3-none-any.whl (109 kB)
  Using cached google_api_core-2.2.2-py2.py3-none-any.whl (95 kB)
  Using cached google_api_core-2.2.1-py2.py3-none-any.whl (95 kB)
  Using cached google_api_core-2.2.0-py2.py3-none-any.whl (95 kB)
  Using cached google_api_core-2.1.1-py2.py3-none-any.whl (95 kB)
  Using cached google_api_core-2.1.0-py2.py3-none-any.whl (94 kB)
  Using cached google_api_core-2.0.1-py2.py3-none-any.whl (92 kB)
  Using cached google_api_core-2.0.0-py2.py3-none-any.whl (92 kB)
  Using cached google_api_core-1.34.0-py3-none-any.whl (120 kB)
  Using cached google_api_core-1.33.2-py3-none-any.whl (115 kB)
  Using cached google_api_core-1.33.1-py3-none-any.whl (115 kB)
  Using cached google_api_core-1.33.0-py3-none-any.whl (115 kB)
  Using cached google_api_core-1.32.0-py2.py3-none-any.whl (93 kB)
  Using cached google_api_core-1.31.6-py2.py3-none-any.whl (93 kB)
  Using cached google_api_core-1.31.5-py2.py3-none-any.whl (93 kB)
  Using cached google_api_core-1.31.4-py2.py3-none-any.whl (93 kB)
  Using cached google_api_core-1.31.3-py2.py3-none-any.whl (93 kB)
  Using cached google_api_core-1.31.2-py2.py3-none-any.whl (93 kB)
  Using cached google_api_core-1.31.1-py2.py3-none-any.whl (93 kB)
  Using cached google_api_core-1.31.0-py2.py3-none-any.whl (93 kB)
Collecting googleapis-common-protos[grpc]<2.0.0dev,>=1.5.2
  Using cached googleapis_common_protos-1.57.1-py2.py3-none-any.whl (218 kB)
  Using cached googleapis_common_protos-1.57.0-py2.py3-none-any.whl (217 kB)
  Using cached googleapis_common_protos-1.56.4-py2.py3-none-any.whl (211 kB)
  Using cached googleapis_common_protos-1.56.3-py2.py3-none-any.whl (211 kB)
  Using cached googleapis_common_protos-1.56.2-py2.py3-none-any.whl (211 kB)
  Using cached googleapis_common_protos-1.56.1-py2.py3-none-any.whl (211 kB)
  Using cached googleapis_common_protos-1.56.0-py2.py3-none-any.whl (241 kB)
  Using cached googleapis_common_protos-1.55.0-py2.py3-none-any.whl (212 kB)
  Using cached googleapis_common_protos-1.54.0-py2.py3-none-any.whl (207 kB)
  Using cached googleapis_common_protos-1.53.0-py2.py3-none-any.whl (198 kB)
  Using cached googleapis_common_protos-1.52.0-py2.py3-none-any.whl (100 kB)
Using legacy 'setup.py install' for hiredis, since package 'wheel' is not installed.
Installing collected packages: yarl, Werkzeug, websocket-client, python-dateutil, proto-plus, packaging, openapi-spec-validator, Jinja2, hiredis, grpcio, googleapis-common-protos, future, frozenlist, ecdsa, docutils, docker-pycreds, defusedxml, Cython, click, charset-normalizer, cffi, certifi, cachetools, attrs, requests, python-jose, libcst, google-crc32c, google-auth, Flask, cryptography, clickclick, botocore, aiosignal, aioredis, s3transfer, grpc-google-iam-v1, google-resumable-media, google-api-core, docker, connexion, azure-storage-common, aiohttp, google-cloud-core, boto3, azure-storage-blob, aiohttp-jinja2, aiohttp-cors, aiohttp-cache, google-cloud-storage, google-cloud-pubsub
  Attempting uninstall: yarl
    Found existing installation: yarl 1.8.2
    Uninstalling yarl-1.8.2:
      Successfully uninstalled yarl-1.8.2
  Attempting uninstall: python-dateutil
    Found existing installation: python-dateutil 2.8.2
    Uninstalling python-dateutil-2.8.2:
      Successfully uninstalled python-dateutil-2.8.2
  Attempting uninstall: packaging
    Found existing installation: packaging 23.0
    Uninstalling packaging-23.0:
      Successfully uninstalled packaging-23.0
  Running setup.py install for hiredis ... error
  error: subprocess-exited-with-error

  × Running setup.py install for hiredis did not run successfully.
  │ exit code: 1
  ╰─> [20 lines of output]
      /tmp/pip-install-lllbjqa9/hiredis_24d9147f61a34e089c14c1fb8ec687fd/setup.py:7: DeprecationWarning: the imp module is deprecated in favour of importlib; see the module's documentation for alternative uses
        import sys, imp, os, glob, io
      /root/wazuh-4.4/lib/python3.9/site-packages/setuptools/dist.py:717: UserWarning: Usage of dash-separated 'description-file' will not be supported in future versions. Please use the underscore name 'description_file' instead
        warnings.warn(
      running install
      running build
      running build_py
      creating build
      creating build/lib.linux-x86_64-3.9
      creating build/lib.linux-x86_64-3.9/hiredis
      copying hiredis/version.py -> build/lib.linux-x86_64-3.9/hiredis
      copying hiredis/__init__.py -> build/lib.linux-x86_64-3.9/hiredis
      running build_ext
      building 'hiredis.hiredis' extension
      creating build/temp.linux-x86_64-3.9
      creating build/temp.linux-x86_64-3.9/src
      creating build/temp.linux-x86_64-3.9/vendor
      creating build/temp.linux-x86_64-3.9/vendor/hiredis
      x86_64-linux-gnu-gcc -Wno-unused-result -Wsign-compare -DNDEBUG -g -fwrapv -O2 -Wall -g -fstack-protector-strong -Wformat -Werror=format-security -g -fwrapv -O2 -g -fstack-protector-strong -Wformat -Werror=format-security -Wdate-time -D_FORTIFY_SOURCE=2 -fPIC -Ivendor -I/root/wazuh-4.4/include -I/usr/include/python3.9 -c src/hiredis.c -o build/temp.linux-x86_64-3.9/src/hiredis.o
      error: command 'x86_64-linux-gnu-gcc' failed: No such file or directory
      [end of output]

  note: This error originates from a subprocess, and is likely not a problem with pip.
error: legacy-install-failure

× Encountered error while trying to install package.
╰─> hiredis

note: This is an issue with the package mentioned above, not pip.
hint: See above for output from the failure.
WARNING: You are using pip version 22.0.4; however, version 23.0.1 is available.
You should consider upgrading via the '/root/wazuh-4.4/bin/python3.9 -m pip install --upgrade pip' command.
```

Pip's log shows that hiredis package has an installation problem. This problem installating the framework's package requirements prevent the installation of other required packages to run the unit tests.

**Comment by mhamra** [comment]
### UPDATE
changing hiredis required version from 1.1.1. to 2.2.2., pip installs all packages without any problem.

hiredis==1.1.1 is an old version, dated 15 Jul 2020. Version 2.2.2 is newer and is dated Feb 13, 2023.

Link to hiredis [CHANGELOG.md](https://github.com/redis/hiredis-py/blob/master/CHANGELOG.md)

**Comment by mhamra** [comment]
### UPDATE
hiredis is an indirecte dependency of aiohttp-cache

```
aiohttp-cache==2.2.0
  - aiohttp [required: >=3.6,<4.0, installed: 3.8.1]
    - aiosignal [required: >=1.1.2, installed: 1.2.0]
      - frozenlist [required: >=1.1.0, installed: 1.2.0]
    - async-timeout [required: >=4.0.0a3,<5.0, installed: 4.0.2]
    - attrs [required: >=17.3.0, installed: 20.3.0]
    - charset-normalizer [required: >=2.0,<3.0, installed: 2.0.4]
    - frozenlist [required: >=1.1.1, installed: 1.2.0]
    - multidict [required: >=4.5,<7.0, installed: 5.1.0]
    - yarl [required: >=1.0,<2.0, installed: 1.6.3]
      - idna [required: >=2.0, installed: 2.9]
      - multidict [required: >=4.0, installed: 5.1.0]
  - aioredis [required: >=1.3,<2.0, installed: 1.3.1]
    - async-timeout [required: Any, installed: 4.0.2]
    - hiredis [required: Any, installed: 2.2.2]
  - envparse [required: >=0.2.0,<0.3.0, installed: 0.2.0]
```

**Comment by davidjiglesias** [comment]
@fdalmaup We should include the update of hiredis in the Python 3.10 upgrade. 

Thanks @mhamra for the investigation!

**Comment by mhamra** [comment]
This issue is addressed by [#16680 - Upgrade hiredis version to >2.1.0](https://github.com/wazuh/wazuh/issues/16680) as part of the epic [#13265 - Upgrade the Wazuh embedded Python to 3.10](https://github.com/wazuh/wazuh/issues/13265)
